packages/react/src/ActionMenu/ActionMenu.tsx

```diff
@@ -0,0 +1,387 @@
+            event.preventDefault()
+            firstItem?.focus()
+          }
+        })
+
+        firstItem?.addEventListener('keydown', event => {
+          if (event.key === 'ArrowUp') {
+            event.preventDefault()
+            lastItem.focus()
+          }
+        })
+
+        //on escape refocus
+        floatingElement.addEventListener('keydown', event => {
+          if (event.key === 'Escape') {
+            anchorElementRef.current?.focus()
+          }
+        })
+      }
+    }, [showMenu, floatingElementRef, handleItemSelection, toggleMenu, handleOnSelect])
+
+    // Calculate the position of the menu
+    const {position} = useAnchoredPosition(
+      {
+        floatingElementRef,
+        anchorElementRef,
+        side: 'outside-bottom',
+        align: menuAlignment,
+        allowOutOfBounds: true
+      },
+      [showMenu]
+    )
+
+    const {Button: SelectButton, Overlay: SelectOverlay} = Children.toArray(children).reduce<{
+      Button?: ReactElement<ActionMenuButtonProps>
+      Overlay?: ReactElement<ActionMenuOverlayProps>
+    }>((acc, child) => {
+      if (isValidElement(child)) {
+        if (child.type === ActionMenuButton) {
+          acc['Button'] = cloneElement(child as ReactElement<ActionMenuButtonProps>, {
+            onClick: toggleMenu,
+            ref: anchorElementRef as React.RefObject<HTMLButtonElement>,
+            className: clsx(child.props.className, showMenu && styles['ActionMenu__button--active']),
+            menuOpen: showMenu,
+            disabled,
+            id: `${instanceId}-button`
+          })
+        } else if (child.type === ActionMenuOverlay) {
+          acc['Overlay'] = cloneElement(child as ReactElement<ActionMenuOverlayProps>, {
+            ref: floatingElementRef as React.RefObject<HTMLUListElement>,
+            className: clsx(styles.ActionMenu__menu, position && styles['ActionMenu__menu--visible']),
+            style: {
+              top: `${position?.top ?? 0}px`,
+              left: `${position?.left ?? 0}px`
+            },
+            id: `${instanceId}-menu`,
+            'aria-labelledby': ariaLabel,
+            children: Children.map(child.props.children, item => {
+              if (isValidElement(item)) {
+                return cloneElement(item as ReactElement<ActionMenuItemProps>, {
+                  handler: handleItemSelection,
+                  type: selectionVariant
+                })
+              }
+              return item
+            })
+          })
+        }
+      }
+      return acc
+    }, {})
+
+    return (
+      <div
+        id={instanceId}
+        className={clsx(styles.ActionMenu, disabled && styles['ActionMenu--disabled'])}
+        data-testid={testId || testIds.root}
+      >
+        {SelectButton}
+        {showMenu ? SelectOverlay : null}
+      </div>
+    )
+  }
+)
+
+type ActionMenuButtonProps = PropsWithChildren<Ref<HTMLButtonElement>> & {
+  'aria-label': string
```

Comment:
Moved over to Overlay, but made it required for reasons we discussed in DMs. 

For posterity, these reasons were:

- aria-label is important for screen reader users. 
- there's a change it could be forgotten if we leave users to add this in
- this is therefore now required at the API-level to use this component